src/app/components/task/DatetimeRespondTask.js

```diff
@@ -228,8 +213,10 @@ class DatetimeRespondTask extends Component {
             name="response"
             value={this.state.date}
             onChange={this.handleChange.bind(this)}
-            okLabel={this.props.intl.formatMessage(messages.ok)}
-            cancelLabel={this.props.intl.formatMessage(messages.cancel)}
+            okLabel={<FormattedMessage id="datetimeRespondTask.ok" defaultMessage="OK" />}
+            cancelLabel={
+              <FormattedMessage id="datetimeRespondTask.cancel" defaultMessage="Cancel" />
+            }
             labelFunc={
               this.state.date ?
                 date => this.props.intl.formatDate(date) : null
```

Comment:
It was not possible to replace `intl.formatDate` by `<FormattedDate />` here?

Reply:
I can't make a risk-vs-reward case for nixing `intl` here.

`@material-ui/pickers` is ... well ... a turd. That's why there's `this.state.date` _and_ `date` in this bit of code: `this.state.date` is _our_ state (which we need). `date` is the picker's state (even though it should be using a prop). We don't know when they differ.

`labelFunc` must return a `String`: a `<FormattedMessage>` isn't allowed. So we have two options:

A) Assume `this.state.date === date` always, and wrap the whole component in a `<FormattedMessage>`
B) Leave the `intl` call.

I see great risk in option A: we'd be trusting the code _we know to be wrong_ to behave the way we expect.

Reply:
Ah, gotcha... thanks!